src/redis-cli.c

```diff
@@ -2051,36 +2056,32 @@ static sds *getSdsArrayFromArgv(int argc, char **argv, int quoted) {
 
 static int issueCommandRepeat(int argc, char **argv, long repeat) {
     while (1) {
+        if (config.cluster_reissue_command || context == NULL ||
+            context->err == REDIS_ERR_IO || context->err == REDIS_ERR_EOF)
+        {
+            if (cliConnect(CC_FORCE) != REDIS_OK) {
+                cliPrintContextError();
+                return REDIS_ERR;
+            }
+        }
         config.cluster_reissue_command = 0;
         if (config.cluster_send_asking) {
             if (cliSendAsking() != REDIS_OK) {
-                if (cliConnect(CC_FORCE) != REDIS_OK) {
-                    return REDIS_ERR;
-                }
-                if (cliSendAsking() != REDIS_OK) {
-                    return REDIS_ERR;
-                }
-            }
-        }
-        if (cliSendCommand(argc,argv,repeat) != REDIS_OK) {
-            cliConnect(CC_FORCE);
-
-            /* If we still cannot send the command print error.
-             * We'll try to reconnect the next time. */
-            if (cliSendCommand(argc,argv,repeat) != REDIS_OK) {
                 cliPrintContextError();
                 return REDIS_ERR;
             }
+            config.cluster_send_asking = 0;
```

Comment:
`config.cluster_send_asking` was unset in `cliSendAsking`. We should remove one. So the problem is if `cliSendAsking` fails, whether we should unset it.

Reply:
i agree.. can you please take over this PR again? test all the various ways this code can be used and make sure it's ready to merge?

the reason i took over was that i flagged it with `to-be-merged` (due to the "ask" parsing) so it showed on my dashboard, i had an intention to clean up repeated calls to cliConnect, but it looked as if you moved on, and i wanted to clean my dashboard.
Then after i already started working on it, we got the report from #9277, which was excellent timing.

Reply:
OK. I will do it.

Reply:
@oranagra It works fine. I only change two lines. I reset `config.cluster_reissue_command` and `config.cluster_send_asking` when connecting failed. I think if connecting failed, we should reset these flags, as user will retry, these flags will be set again if needed.